### PR TITLE
Fix cosmosdb state store to handle content-type, with JSON as default.

### PR DESCRIPTION
**Pull request by artursouza**
# Description

Fix cosmosdb state store to handle content-type, with JSON as default.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #745 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [X] Code compiles correctly
* [X] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Review by pkedy — APPROVED** [review]
LGTM
I made some comments that are sort of a `multipart/nitpick`.  ;)

File: state/azure/cosmosdb/cosmosdb.go [hunk 1]
```diff
@@ -335,23 +339,46 @@ func (c *StateStore) Multi(request *state.TransactionalStateRequest) error {
 	return nil
 }
 
-// This is a helper to return the partition key to use.  If if metadata["partitionkey"] is present,
-// use that, otherwise use what's in "key".
-func populatePartitionMetadata(key string, requestMetadata map[string]string) string {
-	if val, found := requestMetadata[metadataPartitionKey]; found {
-		return val
+func createUpsertItem(contentType string, req state.SetRequest, partitionKey string) CosmosItem {
```
[the remark this answers is not in the document]
Comment (artursouza) [reply state/azure/cosmosdb/cosmosdb.go:342]: Exactly. I tried that path and realized the unit test adds more value here.

File: state/azure/cosmosdb/cosmosdb.go [hunk 2]
```diff
@@ -70,54 +73,55 @@ func NewCosmosDBStateStore(logger logger.Logger) *StateStore {
...
-	connInfo := metadata.Properties
+	connInfo := meta.Properties
 	b, err := json.Marshal(connInfo)
 	if err != nil {
 		return err
 	}
 
-	var creds credentials
-	err = json.Unmarshal(b, &creds)
+	var m metadata
```
[the remark this answers is not in the document]
Comment (artursouza) [reply state/azure/cosmosdb/cosmosdb.go]: I did not want to add the Decode thing here yet. Handled the default the way you suggested.

**Review by pkedy — DISMISSED** [review]

**Review by pkedy — APPROVED** [review]

**Review by rynowak — APPROVED** [review]

File: state/azure/cosmosdb/cosmosdb.go [hunk 3]
```diff
@@ -335,23 +359,46 @@ func (c *StateStore) Multi(request *state.TransactionalStateRequest) error {
...
+			// if byte array is not a valid JSON, so keep it as-is to be Base64 encoded in CosmosDB.
+			// otherwise, we save it as JSON
+			if err == nil {
+				return CosmosItem{
+					ID:           req.Key,
+					Value:        value,
+					PartitionKey: partitionKey,
+					IsBinary:     false,
+				}
+			}
```
Comment (mukundansundar) [inline state/azure/cosmosdb/cosmosdb.go:377]: i think err is being swallowed here. If it is not nil(improper json), it should be returned to the caller?

Comment (artursouza) [reply state/azure/cosmosdb/cosmosdb.go:377]: This is by design. In case the data is not valid JSON, we write it as binary.

**Comment by DenisaJurca** [comment]
Is this fix available in any version of the SDK?

**Pull request merged**